options.go

```diff
@@ -94,3 +94,27 @@ func (o *FetchOptions) Validate() error {
 
 	return nil
 }
+
+// PushOptions describe how a push should be performed.
+type PushOptions struct {
+	// RemoteName is the name of the remote to be pushed to.
+	RemoteName string
+	// RefSpecs specify what destination ref to update with what source
+	// object. A refspec with empty src can be used to delete a reference.
+	RefSpecs []config.RefSpec
+}
+
+// Validate validate the fields and set the default values
+func (o *PushOptions) Validate() error {
```

Comment:
See the above comment, I'm using the same conventions for Validate as in the rest of the code base. The example doesn't use Validate, the user doesn't need to call it, since it's always called by porcelain methods when they get options.